### DISCUSSION TITLE
What is the fastest DPL zero export setup

### DISCUSSION THREAD

**Discussion by oldman4U** [hoylabs/OpenDTU-OnBattery#1230] [Q&A]
I use the DTSU666/Hoymiles DTU Pro and a HM-800 on 16s LiFePO4 battery for zero export control. In case the amount of power used changes, it takes approximately 10sec to adjust to the new value.

What is the fastest/most accurate setup based on OpenDTU?

Sorry if this topic has been discussed before but I could not find something related in the discussions.

Thank you

**Comment by schlimmchen** [comment]
> What is the fastest/most accurate setup based on OpenDTU?

You mean OpenDTU-OnBattery?

One inverter only, latest release installed, power meter polling interval at 1s or even better MQTT power meter interface where the values are published once a second (faster power meters are not really available, AFAIK), DTU polling interval set to 1s. Also, the wireless connection to your inverer *must* be good, i.e., not too far away, not too close, little packet drop. See if the data age in the live view is often larger than 1 second when the DTU interval is 1s, then it should only ever be 1s in the ideal case. Then you can expect your inverter to react in the range of 2 to 3 seconds.

**Reply by greymda** [reply]
@schlimmchen can't find power meter polling interval set location for HTTP(S) + Json option. Can you help me out, please? Cheers

**Reply by schlimmchen** [reply]
That probably means you are on an older firmware release. Is that correct?

**Reply by greymda** [reply]
yes, on a 4mb remote system, [2024.06.03]

**Reply by schlimmchen** [reply]
Well... You can still download/backup your `config.json`, edit the polling interval, then upload the changed config. At least I think I remember that this workaround is viable. Otherwise, you are out of luck, as the feature to set the polling interval through the web UI was only implemented after the fairly recent powermeter refactoring. That's why you don't see the input field:

![image](https://github.com/user-attachments/assets/035cfc83-acd1-4282-b015-f0a2f8cb7bc5)

**Comment by oldman4U** [comment]
Yes. OpenDTU on battery.

Would a Shelly be the preferred meter?

**Reply by gitisgreat2023** [reply]
I have a Shelly pro 3EM. With http://shelly.fritz.box/rpc/Shelly.GetStatus its very fast, only by testing it in the browser it gives sub 1 sec readouts. I have the latest firmware installed though, was delivered with some old version where the readout in the app and web was very slow (didn't test the rpc on the old one though).

Who knows what one can squeeze out of the Hichi WiFi version, read-out speed wise?

**Reply by schlimmchen** [reply]
> Who knows what one can squeeze out of the Hichi WiFi version, read-out speed wise?

That depends on the meter. Mine spits out new values in 1 second intervals (roughly) and the Hichi publishes them immediately.

The Shelly 3 EM also has a measurement frequency of around 1 Hz. And this seems to be typical, as the respective chips have similar performance.

**Reply by gitisgreat2023** [reply]
@schlimmchen I just assembled my Hichi WiFi + Hichi TTL to support both reading out the Hichi and being able to use the Tibber pulse. It works! (Two 3D printed parts put the three modules together, a 3V power supply for the Tibber completes it ;-))
Only: how do I get a fast readout? The lowest value TelePeriod accepts is 10 sec. 
I assume you get the data out to some server, as you write „ Mine spits out new values in 1 second intervals (roughly) and the Hichi publishes them immediately.“

**Reply by spcqike** [reply]
@gitisgreat2023 hichi works with tasmota, doesn’t it? 

If so, check out the wiki 

https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Power-Meter#important-for-tasmota-users

**Reply by schlimmchen** [reply]
Unfortunately, the best approach using a modified `>S` section is not documented there...

```
>S
[...]
=>publish grid/power/total/watts %sml[10]%
```

Where 10 is the 1-based index of the decoded SML value.

I would love to document this properly, but my time is limited, so...

> I assume you get the data out to some server, as you write

@gitisgreat2023 Well yeah, see above (and the other two places where I buried this info^^)

**Reply by Manos1966** [reply]
> I would love to document this properly, but my time is limited, so...
(...the other two places where I buried this info)

any links where to find these Bernhard?

**Comment by oldman4U** [comment]
Another question. Is there a meter which can be connected directly to the esp and has this any further advantages? Thank you all for your input.

**Reply by remigauthier** [reply]
Not sure it will be of any use, the problem will stand between the dtu and the inverters, it is slow : min 1s, a few seconds in real use, sometimes more if a lot of requests are made to the inverters.

**Reply by remigauthier** [reply]
see https://github.com/tbnobody/OpenDTU/issues/1088

**Reply by remigauthier** [reply]
I am using Home Assistant with slow zero export using Automation rules right now.

I reduce power of each inverter from 100 to 50 to 5% Non Persistent Relative Limit one inverter after the other as long as I export power  and I increase power same way when I import power --> 10s interval between each action.

As a result I export power and there are spikes but overall the goal is reached, see purple to see exported power in the graph shared.

![Capture d'écran 2024-09-29 002330](https://github.com/user-attachments/assets/540a5384-6ae0-49c2-a971-4ac404cc114d)
